tensorboard/plugins/base_plugin.py

```diff
@@ -32,16 +32,21 @@ class TBPlugin(object):
   """TensorBoard plugin interface. Every plugin must extend from this class.
 
   Subclasses must have a constructor that takes a TBContext argument.
+
+  Fields:
+    plugin_name: The plugin_name will also be a prefix in the http handlers 
+        generated by the plugin, e.g. `data/plugins/$PLUGIN_NAME/$HANDLER` The 
+        plugin name must be unique for each registered plugin, or a ValueError 
+        will be thrown when the application is constructed. The plugin name must 
+        only contain characters among [A-Za-z0-9_.-], and must be nonempty, 
+        or a ValueError will similarly be thrown.
+    disable_path_prefix: If True then data/plugins/$PLUGIN_NAME prefix will not
```

Comment:
I agree with William here - there's no reason to allow plugins to disable the path prefix, it is just an invitation for trouble.
I also agree that changing logdir and runs to come from data/plugin/core/logdir. If we are moving everything into plugins, we could likely shorten the "data/plugin/" prefix (e.g. just to "plugin") but there's no need to, either.